nncf/experimental/netron.py

```diff
@@ -0,0 +1,197 @@
+
+        if self.inputs:
+            input = ET.SubElement(node, Tags.INPUT)
+            for port in self.inputs:
+                input.append(port.as_xml_element())
+
+        if self.outputs:
+            output = ET.SubElement(node, Tags.OUTPUT)
+            for port in self.outputs:
+                output.append(port.as_xml_element())
+
+        return node
+
+
+class EdgeDesc:
+
+    def __init__(self,
+                 from_node: str,
+                 from_port: str,
+                 to_node: str,
+                 to_port: str):
+        self.from_node = from_node
+        self.from_port = from_port
+        self.to_node = to_node
+        self.to_port = to_port
+
+    def as_xml_element(self) -> ET.Element:
+        attrs = {
+            'from-layer': self.from_node,
+            'from-port': self.from_port,
+            'to-layer': self.to_node,
+            'to-port': self.to_port,
+        }
+        edge = ET.Element(Tags.EDGE, attrs)
+        return edge
+
+
+# TODO(andrey-churkin): Add support for `PortDesc.precision` param.
+def get_graph_desc(graph: NNCFGraph,
+                   include_fq_params: bool = False) -> Tuple[List[NodeDesc], List[EdgeDesc]]:
+    include_node = {}  # Dict[int, bool]
+    edges = []
+    for edge in graph.get_all_edges():
+        if not include_fq_params and edge.to_node.node_type == 'FakeQuantize' and edge.input_port_id != 0:
+            include_node[edge.from_node.node_id] = False
+            continue
+
+        edges.append(
+            EdgeDesc(
+                from_node=str(edge.from_node.node_id),
+                from_port=str(edge.output_port_id),
+                to_node=str(edge.to_node.node_id),
+                to_port=str(edge.input_port_id)
+            )
+        )
+
+    nodes = []
+    for node in graph.get_all_nodes():
+        if not include_node.get(node.node_id, True):
+            continue
+
+        inputs = []
+        for edge in graph.get_input_edges(node):
+            if not include_fq_params and node.node_type == 'FakeQuantize' and edge.input_port_id != 0:
+                continue
+
+            inputs.append(
+                PortDesc(
+                    port_id=str(edge.input_port_id),
+                    shape=edge.tensor_shape
+                )
+            )
+
+        outputs = []
+        for edge in graph.get_output_edges(node):
+            outputs.append(
+                PortDesc(
+                    port_id=str(edge.output_port_id),
+                    shape=edge.tensor_shape,
+                )
+            )
+
+        nodes.append(
+            NodeDesc(
+                node_id=str(node.node_id),
+                name=node.node_name,
+                type=node.node_type,
+                attrs={
+                    'metatype': str(node.metatype.name),
+                },
+                inputs=inputs,
+                outputs=outputs
+            )
+        )
+
+    return nodes, edges
+
+
+def save_for_netron(graph: NNCFGraph,
```

Comment:
Could this be the NNCFGraph method? The user must have NNCFGraph object to call `save_for_netron`, so it looks like easily to call `graph.save(...) ` instead of `nncf.*.save_for_netron(graph, ...)`

Reply:
Do we really need to show the `NNCFGraph` object to the user? May be it is better do not show this object and use code like this
```python
quantized_model = nncf.quantize(model, calibration_dataset)
nncf.save_graph_for_netron(quantized_model, 'quantized_model.xml')
```

Reply:
If I understand correctly, the user only needs the NNCFGraph to save it.

Reply:
@AlexKoff88 What do you think about how we should expose this method to users?

Reply:
I would agree that it is better to have it as a method of the wrapped object:
```python
quantized_model.save_for_netron("quantized_model.xml")
```